db/db.go

```diff
@@ -81,6 +82,19 @@ func (t *TinkDB) Migrate() (int, error) {
 	return migrate.Exec(t.instance, "postgres", migration.GetMigrations(), migrate.Up)
 }
 
+// WithMigrations applies only the given migrations.
+// It is required to unit test migrations.
+func (t *TinkDB) WithMigrations(migrations []func() *migrate.Migration) error {
```

Comment:
I don't think we need this function for mainly one reason. This is a new API that should not be used outside of `tests`. So it is in the wrong place, it should be exposed via `TinkDB` because it allows developers to use it outside of tests, and we do not want that.

Another reason is that it does not add much value or simplification you use it this way:

```go
	err := tinkDB.WithMigrations([]func() *migrate.Migration{
		migration.Get202009171251,
		migration.Get202010071530,
		migration.Get202010221010,
		migration.Get202012041103,
		migration.Get202012091055,
		migration.Get2020121691335,
	})
```

But you can get to the same point writing:

```go
	_, err := migrate.Exec(db, "postgres", migrate.MemoryMigrationSource{
		Migrations: []*migrate.Migration{
			migration.Get202009171251(),
			migration.Get202010071530(),
			migration.Get202010221010(),
			migration.Get202012041103(),
			migration.Get202012091055(),
			migration.Get2020121691335(),
		},
	}, migrate.Up)
```

Almost same line of code. If you are wondering where `db` is coming from:

```go
db, tinkDB, cl := NewPostgresDatabaseClient(t, ctx, NewPostgresDatabaseRequest{})
```